kotlin/edit-with-branch-versioning/src/main/java/com/esri/arcgisruntime/sample/editwithbranchversioning/MainActivity.kt

```diff
@@ -0,0 +1,354 @@
+import com.esri.arcgisruntime.security.DefaultAuthenticationChallengeHandler
+import kotlinx.android.synthetic.main.activity_main.*
+
+class MainActivity : AppCompatActivity() {
+
+  private val TAG = MainActivity::class.java.simpleName
+
+  private val serviceGeodatabase: ServiceGeodatabase by lazy {
+    ServiceGeodatabase("https://sampleserver7.arcgisonline.com/arcgis/rest/services/DamageAssessment/FeatureServer").apply {
+      loadAsync()
+    }
+  }
+
+  private var featureLayer: FeatureLayer? = null
+  private var selectedFeature: Feature? = null
+  private var createdVersionName: String = ""
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    setContentView(R.layout.activity_main)
+
+    // create a default authentication challenge handler because this service requires authentication
+    AuthenticationManager.setAuthenticationChallengeHandler(
+      DefaultAuthenticationChallengeHandler(
+        this
+      )
+    )
+
+    // create a map with a streets vector basemap and set it to the mapview
+    mapView.map = ArcGISMap(Basemap.createStreetsVector())
+
+    serviceGeodatabase.addDoneLoadingListener {
+      // check if the service geodatabase failed to load
+      if (serviceGeodatabase.loadStatus != LoadStatus.LOADED) {
+        serviceGeodatabase.loadError?.let {
+          Log.e(TAG, "Service Geodatabase failed to load: ${it.cause}")
+        }
+        return@addDoneLoadingListener
+      }
+      // create a feature layer from the service geodatabase's service feature table
+      val serviceFeatureTable = serviceGeodatabase.getTable(0)
+      featureLayer = FeatureLayer(serviceFeatureTable).apply {
+        addDoneLoadingListener {
+          // zoom to the layer's extent
+          mapView.setViewpointAsync(Viewpoint(fullExtent))
+        }
+      }
+      // add the layer to the map
+      mapView.map.apply {
+        operationalLayers.add(featureLayer)
+      }
+      // display the current name in a text view
+      currentVersionNameTextView.text = serviceGeodatabase.versionName
+
+      mapView.onTouchListener = object : DefaultMapViewOnTouchListener(this, mapView) {
+        override fun onSingleTapConfirmed(e: MotionEvent): Boolean {
+          // The default version, sde.DEFAULT, is protected and the user provided
+          // is not the owner so edits will not be allowed.
+          // So for simplicity, if the current version is the default, prevent editing
+          if (serviceGeodatabase.versionName.isBlank() || serviceGeodatabase.versionName == serviceGeodatabase.defaultVersionName) {
+            val message = "This sample does not allow editing of features on the default version."
+            Log.e(TAG, message)
+            Toast.makeText(this@MainActivity, message, Toast.LENGTH_LONG).show()
+            return true
+          }
+
+          val point = android.graphics.Point(e.x.toInt(), e.y.toInt())
```

Comment:
maybe just a comment on this line or distinguish this is a screenPoint rather than a mapPoint